test/test-cid.js

```diff
@@ -472,7 +506,7 @@ describe('CID', () => {
       const hash = await sha256.digest(textEncoder.encode('abc'))
       const cid = CID.create(1, 112, hash)
 
-      const parsed = CID.parse(cid.toString(base58btc))
+      const parsed = /** @type {CID} */(CID.parse(cid.toString(base58btc)))
```

Comment:
any idea why it's failing to infer this?

Reply:
It's not failing it's just return type is `null|CID` this manually refines it to `CID`, which is something TS allows.